src/til/ut_til/StaticMapTests.cpp

```diff
@@ -95,4 +95,29 @@ class StaticMapTests
         VERIFY_THROWS(unused = intIntMap[7], std::runtime_error);
     }
 #pragma warning(pop)
+
+    TEST_METHOD(Presort)
+    {
+        static constexpr til::presorted_static_map intIntMap{
+            std::pair{ 1, 100 },
+            std::pair{ 3, 300 },
+            std::pair{ 5, 500 },
+        };

```

Comment:
So, what happens for the following two scenarios:
```
        // 1. sorted, but backwards
        static constexpr til::presorted_static_map intIntMap{
            std::pair{ 5, 500 },
            std::pair{ 3, 300 },
            std::pair{ 1, 100 },
        };

        // 2. not sorted, but declared as presorted
        static constexpr til::presorted_static_map intIntMap{
            std::pair{ 3, 300 },
            std::pair{ 1, 100 },
            std::pair{ 5, 500 },
        };
```

Reply:
Read the comment in the header!

Reply:
```
// A failure to sort your keys will result in unusual
// runtime behavior, but no error messages will be
// generated.
```
ah...and I'm guessing reverse doesn't count :(

Reply:
Kinda wish it were possible to throw a compiler error if they weren't sorted...

Reply:
Alas! That might require constexpr sort!

Unless i can do constexpr “check if the predicate is false” for every element pair ??

Reply:
yeah there might be no good option here. I'm gonna sign it because it is probably worth the risk to hold us over until the real goodness arrives.